### DISCUSSION TITLE
Android - Compass in Precise View not working correctly

### DISCUSSION THREAD

**Discussion by dewey-b** [opengisch/QField#4254] [Bugs]
I have a Pixel 7 Pro and a Pixel 6a Android phone. I get the following identical results with each phone.

When I select a feature, and set the feature as destination, the compass in “Precise View” points in random directions. When I rotate my phone, the direction of North changes to a different direction that does not seem to be related to north or the orientation of my phone. The Bearing (to feature) is at the proper degrees on the compass, but because north on the compass does not point to north, the bearing points in the wrong direction. Therefore, I can not use Precise View to pinpoint the feature. Other compass applications on my phone work correctly. Is this a bug or have I missed a setting?

If I could lock the compass with north up, this would be an improvement. This is how Qfield works on my Windows laptop that does not have a compass. Is there a setting to lock north up?

**Comment by Catalin75** [comment]
Good afternoon ! I have the same issue on one of my tablets. I think there is an option to be activated, but I just cannot remember which it is. On an older tablet, with same version of QFIELD installed, (the latest), the compass in Precise View (the trigonometrical circle) is steady, showing always north in the upper part, which makes the staking out of a point (feature) very easy. Maybe someone could indicate which setting should be activated. Thank you in advance ! Have a nice day !

**Comment by Catalin75** [comment]
After some searching, I found out that by turning off the sensors on your Android phone, the issue is solved. The Precise View is showing always north, steady, and the staking out of a point is much easier. Have a nice day !

**Comment by nirvn** [comment]
@dwightbrew , can you provide a video as to better understand what you see?

**Reply by Catalin75** [reply]
https://github.com/opengisch/QField/assets/134632643/e96eb268-fe07-4a34-a3a4-de742c401855


https://github.com/opengisch/QField/assets/134632643/8cdbed4a-798c-4ac5-9704-ccd8969944c0

**Comment by dewey-b** [comment]
I upgraded to QField 2.8.2 a couple of weeks ago and that seems to have made things better, but not perfect.

If the QField compas points to true north (which I suspect it does), then I think I am only seeing a small discrepancy when I rotate my phone. When I made my original post, I was running an earlier version of QField, and the difference was much greater.

I am attaching 5 files.

1. setup.jpg - this shows my setup
2. compass.jpg - this shows magnetic north and is aligned with my blue arrow which does not move.
3. QField 1.jpg, 4. QField 2.jpg, and 5. QField 3.jpg - each image is taken with the exact same setup. The only difference is that the phone is rotated between shots.

![1  setup](https://github.com/opengisch/QField/assets/10574893/f6f0e9b7-94f1-40f3-810b-98c431256f4e)
![2  compass](https://github.com/opengisch/QField/assets/10574893/469240f1-0272-47d9-907c-94ae4e70f3a3)
![3  QField 1](https://github.com/opengisch/QField/assets/10574893/62873147-54b1-4fa8-9423-6a6df1fe05dd)
![4  QField 2](https://github.com/opengisch/QField/assets/10574893/d9e349f8-f3dc-441f-a384-9114997c1070)
![5  QField 3](https://github.com/opengisch/QField/assets/10574893/04cd275b-e74b-4acd-ab06-62c5cb53cfeb)

**Reply by nirvn** [reply]
@dwightbrew , this ought to win a "best follow details post" of the year, you've just made my weekend :wink: 

One test I'd be most interested in you doing is whether you _facing_ the destination point with your phone results in the precise view rotating so that the destination blue dot (or the blue arrow when you have precise view set to always show) is aligned at the top. That's ultimately what this rotation thing is about, it is meant to help you rotate yourself (using your mobile device's magnetometer) towards the target in a straight line.

We're using magnetic north as provided by the phone. I happen to have a pixel 6a too, and it works well over here. Which CRS are you using for your project?

**Reply by nirvn** [reply]
@dwightbrew , can you try the APKs in this pull request: https://github.com/opengisch/QField/pull/4378 -- I've tried to tweak some compass / orientation handling based on CRSes bearing to true north, it might result in a better experience for you.

**Reply by nirvn** [reply]
@dwightbrew , gentle ping, have you had the opportunity to test?

**Reply by dewey-b** [reply]
@nirvn , I have both v2.8.2 (f63bd2) and PR4378 (19a9f0) on my phone and they both seem to be working similarly and correctly. When I am far away (30 meters) from the destination, the phone arrow (or circle, I can't remember which) pointing to the destination is slightly off from the destination (perhaps 5 or 10 degrees), but if I follow the arrow or circle I can easily get to the exact destination point. I wouldn't call this an issue. I believe my problem was resolved by either an earlier upgrade of QField or Android 13.

I would be happy to try an earlier version of QField (perhaps the version that was in production on May 2) if you would like. I would need a link to that version for the test.
